src/components/Camera/index.js

```diff
@@ -121,13 +141,15 @@ export default class Camera extends React.Component<CameraType, CameraStateType>
   }
 
   handleWebcamFailure = () => {
-    this.setState({ hasGrantedPermission: false })
+    if (this.state.hasGrantedPermission) {
```

Comment:
Yes, we could. Would you suggest inspecting the error to see if it's a permission error?

My idea was based on this comment from @seewah:
> it is basically to catch any sort of webcam error ONCE permission has been granted. in these cases, we don't worry too much helping users recover. instead we will just tell them to retry

Reply:
I see, the thing is that on some browser you might actually get a permission denied after the permission had been granted before, right?
How consistent is the permission denied error in different browsers?

Reply:
Yes, this issue is related to the one discussed [here](https://github.com/onfido/react-webcam/pull/21#pullrequestreview-134957326). When the webcam fails with one of those errors within the `permissionsErrors` array, the permission denied recovery screen is shown instead of `CAMERA_NOT_WORKING` error. Tested on Huawei EMUI 4.0.3

Reply:
Also discussed [here](https://github.com/onfido/react-webcam/pull/21#discussion_r200658884), `NotAllowedError`, `NotFoundError` and `PermissionDeniedError` seem to be pretty consistent across browsers / OSs.